paasta_tools/api/views/instance.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+# Copyright 2015-2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+PaaSTA service instance status/start/stop etc.
+"""
+import json
+import logging
+import traceback
+
+from pyramid.response import Response
+from pyramid.view import view_config
+
+from paasta_tools import marathon_tools
+from paasta_tools.api import settings
+from paasta_tools.cli.cmds.status import get_actual_deployments
+from paasta_tools.marathon_serviceinit import compose_marathon_job_instance_count
+from paasta_tools.marathon_serviceinit import deploy_status_marathon_job
+from paasta_tools.marathon_serviceinit import get_bouncing_status
+from paasta_tools.utils import NoDockerImageError
+from paasta_tools.utils import PaastaColors
+from paasta_tools.utils import validate_service_instance
+
+log = logging.getLogger(__name__)
+
+
+def chronos_instance_status(instance_status, service, instance, verbose):
+    return
+
+
+def marathon_job_status(client, job_config):
+    mstatus = {}
+
+    try:
+        app_id = job_config.format_marathon_app_dict()['id']
+        mstatus['app_id'] = app_id
+    except NoDockerImageError:
+        error_msg = "Docker image is not in deployments.json. Has Jenkins deployed it?"
+        mstatus['message'] = error_msg
+        return mstatus
+
+    if marathon_tools.is_app_id_running(app_id, client):
+        app = client.get_app(app_id)
+        mstatus['deploy_status'] = PaastaColors.decolor(
```

Comment:
I'm +1 to @keshavdv's ideas. The output formatting (adding a color) should absolutely be done by the caller (and so moved to [here](https://github.com/Yelp/paasta/pull/589/files#diff-417daa104166f0376fb5b1320730cafbR142)).

I also +1 to the idea of using an enum to represent states. As I understand it, right now, the value of the ``status`` key in the output could be:

- 'Running'
- 'Stopped'
- 'Critical'
- 'Waiting (new tasks are not launching due to lack of capacity)'
- 'Delayed (next task won't launch for %s seconds due to previous failures)'

This is pretty inconsistent (why do only some of them deserve a further explanation)? Instead, I think these should be:

- RUNNING
- STOPPED
- CRITICAL
- WAITING
- DELAYED

There should also always be the ``delay`` and ``waiting`` keys printed, irrespective of whether their values are currently impacting the deploy.

I'm not even convinced that DELAYED and WAITING should be states - instead they should be separate booleans returned in the output - the client would decide that we were out of resources by looking at a combination of ``status: CRITICAL`` and ``waiting: True``. I'm not set on this though - just an idea.

> there is not much difference between returning "running" and returning AppStatus.RUNNING in deploy_status_marathon_job

Sure, they're just strings, but the enum provide a consistent interface to whatever those strings may be, reducing the chance of an inconsistency in the codebase ('Deploying' vs 'deploying' vs 'delpoying'). You can see examples of us doing this in [chronos_tools](https://github.com/Yelp/paasta/blob/master/paasta_tools/chronos_tools.py#L67-L74). It's the same reason we use constants.